internal/service/eks/node_group.go

```diff
@@ -313,8 +313,8 @@ func resourceNodeGroupCreate(ctx context.Context, d *schema.ResourceData, meta i
 		input.DiskSize = aws.Int64(int64(v.(int)))
 	}
 
-	if v := d.Get("instance_types").(*schema.Set); v.Len() > 0 {
-		input.InstanceTypes = flex.ExpandStringSet(v)
+	if v := d.Get("instance_types").([]interface{}); len(v) > 0 {
+		input.InstanceTypes = flex.ExpandStringList(v)
```

Comment:
To account for empty values: 
```suggestion
	if v, ok := d.GetOk("instance_types"); ok && len(v.([]interface{})) > 0 && v.([]interface{})[0] != nil {
		input.InstanceTypes = flex.ExpandStringList(v.([]interface{}))
```